plugin/pkg/global/extensionvalidation/admission.go

```diff
@@ -272,17 +274,22 @@ func (e *ExtensionValidator) validateSeed(kindToTypesMap map[string]sets.Set[str
 	return requiredExtensions.areRegistered(kindToTypesMap)
 }
 
-func (e *ExtensionValidator) validateShoot(kindToTypesMap map[string]sets.Set[string], spec core.ShootSpec) error {
+func (e *ExtensionValidator) validateShoot(kindToTypesMap map[string]sets.Set[string], supportedForWorkerlessMap map[string]sets.Set[string], spec core.ShootSpec, workerless bool) error {
 	var (
-		message         = "given Shoot uses non-registered"
-		providerTypeMsg = fmt.Sprintf("%s provider type: %s", message, field.NewPath("spec", "provider", "type"))
-
-		requiredExtensions = requiredExtensions{
-			{extensionsv1alpha1.ControlPlaneResource, spec.Provider.Type, providerTypeMsg},
-			{extensionsv1alpha1.InfrastructureResource, spec.Provider.Type, providerTypeMsg},
-			{extensionsv1alpha1.WorkerResource, spec.Provider.Type, providerTypeMsg},
-		}
+		message            = "given Shoot uses non-registered"
+		providerTypeMsg    = fmt.Sprintf("%s provider type: %s", message, field.NewPath("spec", "provider", "type"))
+		requiredExtensions = requiredExtensions{}
+		result             error
 	)
+
+	if !workerless {
+		requiredExtensions = append(requiredExtensions,
+			requiredExtension{extensionsv1alpha1.ControlPlaneResource, spec.Provider.Type, providerTypeMsg},
+			requiredExtension{extensionsv1alpha1.InfrastructureResource, spec.Provider.Type, providerTypeMsg},
+			requiredExtension{extensionsv1alpha1.WorkerResource, spec.Provider.Type, providerTypeMsg},
+		)
+	}
+
 	if spec.Networking != nil && spec.Networking.Type != nil {
 		requiredExtensions = append(requiredExtensions, requiredExtension{extensionsv1alpha1.NetworkResource, *spec.Networking.Type, fmt.Sprintf("%s networking type: %s", message, field.NewPath("spec", "networking", "type"))})
 	}
```

Comment:
Should we move this also to the above `if` to prevent a misleading error message in case the user mentions a non-supported type? The static validation error will only come once the admission plugins allow the request.